src/main/public/js/postcode-lookup.js

```diff
@@ -0,0 +1,261 @@
+        errorLink.href = '#' + error.dataset.label
+        errorLink.textContent = error.textContent.trim()
+
+        var globalError = document.createElement("li")
+        globalError.appendChild(errorLink)
+        globalErrors.appendChild(globalError)
+      }
+    }
+
+    function clearErrors (global) {
+      widget.querySelectorAll('.form-group-error,.form-control-error').forEach(function (element) {
+        element.classList.remove('form-group-error', 'form-control-error')
+      })
+      addressContainer.querySelectorAll('.error-message').forEach(function (element) {
+        element.remove()
+      })
+      widget.querySelectorAll('.js-error').forEach(function (element) {
+        hide(element)
+      })
+
+      if (global) {
+        var globalErrors = document.querySelector('.error-summary-list')
+        addressContainer.querySelectorAll('label').forEach(function (label) {
+          globalErrors.querySelector('li a[href="#' + label.id  + '"]').remove()
+        })
+      }
+    }
+
+    function setAddress (address) {
+      function getValueOrEmpty(property) {
+        if (address === undefined) {
+          return ''
+        }
+        return address[property]
+      }
+
+      addressContainer.querySelector('.address-line1').value = getValueOrEmpty('line1')
+      addressContainer.querySelector('.address-line2').value = getValueOrEmpty('line2')
+      addressContainer.querySelector('.address-town-or-city').value = getValueOrEmpty('city')
+      addressContainer.querySelector('.postcode').value = getValueOrEmpty('postcode')
+    }
+
+    function setUserInterfaceState (state) {
+      switch (state) {
+        case State.PRISTINE:
+          hide(addressDropdownContainer)
+          hide(addressContainer)
+          setAddress(undefined)
+          show(enterAddressManuallyLink)
+          break
+        case State.ADDRESS_FOUND:
+          show(addressDropdownContainer)
+          break
+        case State.ADDRESS_ENTERED:
+          show(addressContainer)
+          hide(enterAddressManuallyLink)
+          break
+      }
+
+      stateHolder.value = state
+    }
+
+    function lookupPostcode (postcode) {
+      if (!postcode || postcode.trim().length === 0) {
+        showError(Error.MISSING_POSTCODE_ERROR)
+        return
+      }
+
+      var xhr = new XMLHttpRequest()
+      xhr.open('GET', '/postcode-lookup?postcode=' + encodeURIComponent(postcode))
+      xhr.onload = function () {
+        if (xhr.status !== 200) {
+          showError(Error.UNKNOWN_ERROR)
+          return
+        }
+
+        var result = JSON.parse(xhr.responseText)
+
+        if (!result.valid) {
+          function isNorthernIrelandPostcode (postcode) {
+            return postcode.startsWith('BT')
+          }
+
+          showError(isNorthernIrelandPostcode(postcode) ? Error.UNSUPPORTED_POSTCODE_ERROR : Error.UNKNOWN_ERROR)
+          return
+        }
+
+        var nonSelectableOption = document.createElement("option")
```

Comment:
not sure exactly how it is happening but the list isn't being cleared on multiple searches of the same postcode,
![image](https://user-images.githubusercontent.com/21194782/34660820-43262128-f43d-11e7-921e-20e685042353.png)

Note have't investigated too closely, everything works but it doesn't look great.
REC to fix here, could be follow up

Reply:
Good spot, although the same problem exists with the implementation that has been merged into master :(